task/OSMPreprocessing.js

```diff
@@ -0,0 +1,177 @@
+const fs = require('fs');
+const { once } = require('events');
+const readline = require('readline');
+const fse = require('fs-extra');
+const exec = require('child_process').exec;
+const through = require('through2');
+const {
+  dataDir,
+  constants,
+  dataToolImage,
+  osmPreprocessingURLs,
+} = require('../config');
+const { postSlackMessage, createDir } = require('../util');
+
+async function readPreprocessingInstructions(preprocessingInstructionsFile) {
+  const preprocessingInstructions = [];
+  const errorLines = [];
+  const rl = readline.createInterface({
+    input: fs.createReadStream(preprocessingInstructionsFile),
+  });
+  rl.on('line', line => {
+    if (/^(osmconvert|osmfilter|osmupdate).*$/.test(line)) {
+      preprocessingInstructions.push(line);
+    } else if (line === '') {
+      process.stdout.write(
+        'Skipping empty line in ' + preprocessingInstructionsFile + '\n',
+      );
+    } else {
+      errorLines.push(line);
+    }
+  });
+  await once(rl, 'close');
+  if (errorLines.length > 0) {
+    throw Error(
+      `${errorLines.join(', ')} are not valid preprocessing instructions!\n`,
+    );
+  }
+  return `-c '${preprocessingInstructions.join(' && ')}'`;
+}
+
+/**
+ * Runs the instructions listed in the OSM preprocessing file.
+ * Only the osmfilter, osmconvert, and osmupdate commands can be used.
+ */
+function preprocessWithFile(
+  osmFile,
+  quiet = false,
+  osmPreprocessingDlDir,
+  osmId,
+  osmFileName,
+) {
+  const lastLog = [];
+
+  return new Promise((resolve, reject) => {
+    const preprocessingInstructionsFile = `${osmPreprocessingDlDir}/${osmId}.txt`;
+
+    if (!fs.existsSync(osmFile)) {
+      reject(new Error(`${osmFile} does not exist!\n`));
+    } else if (!fs.existsSync(preprocessingInstructionsFile)) {
+      reject(new Error(`${preprocessingInstructionsFile} does not exist!\n`));
+    } else {
+      createDir(`${dataDir}/tmp`);
+      fs.mkdtemp(`${dataDir}/tmp/osm-preprocessing`, (err, folder) => {
+        if (err) throw err;
+        process.stdout.write(
+          'Running OSM preprocessing instructions from ' +
+            osmFile +
```

Comment:
This looks wrong - command are not from osm file.

Reply:
I fixed this